easybuild/easyblocks/i/imkl.py

```diff
@@ -71,6 +71,38 @@ def __init__(self, *args, **kwargs):
         # make sure $MKLROOT isn't set, it's known to cause problems with the installation
         self.cfg.update('unwanted_env_vars', ['MKLROOT'])
 
+        # build the mkl interfaces, if desired
+        if self.cfg['interfaces']:
+            self.cdftlibs = ['fftw2x_cdft']
+            if LooseVersion(self.version) >= LooseVersion('10.3'):
+                self.cdftlibs.append('fftw3x_cdft')
+            self.mpi_spec = None
+            # check whether MPI_FAMILY constant is defined, so mpi_family() can be used
+            if hasattr(self.toolchain, 'MPI_FAMILY') and self.toolchain.MPI_FAMILY is not None:
+                mpi_spec_by_fam = {
+                    toolchain.MPICH: 'mpich2',  # MPICH is MPICH v3.x, which is MPICH2 compatible
+                    toolchain.MPICH2: 'mpich2',
+                    toolchain.MVAPICH2: 'mpich2',
+                    toolchain.OPENMPI: 'openmpi',
+                }
+                mpi_fam = self.toolchain.mpi_family()
+                self.mpi_spec = mpi_spec_by_fam.get(mpi_fam)
+                debugstr = "MPI toolchain component"
+            else:
+                # can't use toolchain.mpi_family, because of dummy toolchain
+                if get_software_root('MPICH2') or get_software_root('MVAPICH2'):
+                    self.mpi_spec = 'mpich2'
+                elif get_software_root('OpenMPI'):
+                    self.mpi_spec = 'openmpi'
+                elif not get_software_root('impi'):
```

Comment:
@bartoldeman At this point, the modules for the dependencies are not loaded yet (that's done in `prepare_step`), so each of these `get_software_root` calls will return `None`...

We should initialize `self.cdftlibs` and `self.mpi_spec` in `__init__`, and move this whole block to `prepare_step`, after the `super(EB_imkl, self).prepare_step` call